plugins/source/azure/client/client.go

```diff
@@ -143,7 +150,10 @@ func (c *Client) Logger() *zerolog.Logger {
 }
 
 func (c *Client) ID() string {
-	return c.SubscriptionId
+	if c.ResourceGroup != "" {
```

Comment:
Follows the Azure convention, see https://github.com/cloudquery/cloudquery/blob/8d9711567ed8eca5e814a1e8840d25923f190bc6/plugins/source/azure/client/helpers.go#L11